### PR TITLE
edl also installs brew libusb

### PR DESCRIPTION
**Pull request by andiradulescu**
For:
- https://github.com/commaai/openpilot/issues/33016 
- https://discord.com/channels/469524606043160576/1263672324973138033

Keeping as draft until I know for sure it works on Apple Silicon.

### CODE REVIEW

File: tools/edl [hunk 1]
```diff
@@ -30,15 +30,40 @@ if [ "$(< .git/HEAD)" != "$VERSION" ]; then
...
-#
-################################################################################
+  # install libusb
+  if [ ! -f $BREW_PREFIX/lib/libusb-1.0.0.dylib ]; then
+    echo "Installing libusb"
+    brew install libusb
+    if [[ $ARCH == "arm64" ]]; then
+      # https://github.com/pyusb/pyusb/issues/520
+      # https://github.com/pyusb/pyusb/issues/355#issuecomment-974726078
+      ln -s $BREW_PREFIX/lib/libusb-1.0.0.dylib /usr/local/lib/libusb.dylib
```
Comment (robin-reckmann) [inline tools/edl]: /usr/local/lib/ does not exist on my M3 Mac. So you would have to create that folder beforehand.

File: tools/edl [hunk 2]
```diff
@@ -30,15 +30,40 @@ if [ "$(< .git/HEAD)" != "$VERSION" ]; then
...
-#
-################################################################################
+  # install libusb
+  if [ ! -f $BREW_PREFIX/lib/libusb-1.0.0.dylib ]; then
+    echo "Installing libusb"
+    brew install libusb
+    if [[ $ARCH == "arm64" ]]; then
+      # https://github.com/pyusb/pyusb/issues/520
+      # https://github.com/pyusb/pyusb/issues/355#issuecomment-974726078
+      sudo ln -s $BREW_PREFIX/lib/libusb-1.0.0.dylib /usr/local/lib/libusb.dylib
```
Comment (robin-reckmann) [inline tools/edl:64]:
```suggestion
      sudo mkdir -p /usr/local/lib
      sudo ln -s $BREW_PREFIX/lib/libusb-1.0.0.dylib /usr/local/lib/libusb.dylib
```

Comment (robin-reckmann) [reply tools/edl:64]: /usr/local/lib/ does not exist on my M3 Mac. So you would have to create that folder beforehand.

Comment (andiradulescu) [reply tools/edl:65]: thanks! I don’t have an M-series mac anymore.

does it work fine for you after this change?

Comment (robin-reckmann) [reply tools/edl:65]: Yes, it works.

File: tools/edl [hunk 3]
```diff
@@ -30,15 +30,41 @@ if [ "$(< .git/HEAD)" != "$VERSION" ]; then
 fi
 popd > /dev/null
 
-$EDL_PATH/edl "$@"
+if [[ "$(uname)" == 'Darwin' ]]; then
+  ARCH=$(uname -m)
+
+  # install brew if required
```
Comment (adeebshihadeh) [inline tools/edl:36]: I know openpilot does it, but installing brew is pretty invasive (plus we don't want to have maintain this). Let's just put "fresh" setup instructions in the README

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

**Comment by andiradulescu** [comment]
Will update the README in another PR.

**Pull request closed without merging**